exec/slicemachine.go

```diff
@@ -549,6 +551,13 @@ func (m *machineManager) Do(ctx context.Context) {
 				have/m.machprocs, have, pending/m.machprocs, pending)
 			go func() {
 				machines := startMachines(ctx, m.b, m.group, m.machprocs, needMachines, m.worker, m.params...)
+				for _, mach := range machines {
+					m.machinesWG.Add(1)
+					go func(machine *sliceMachine) {
+						machine.Go(ctx)
+						m.machinesWG.Done()
+					}(mach)
+				}
```

Comment:
```suggestion
				for _, machine := range machines {
					machine := machine
					m.machinesWG.Add(1)
					go func() {
						machine.Go(ctx)
						m.machinesWG.Done()
					}()
				}
```
It's idiomatic to redeclare. See this [section in Effective Go](https://golang.org/doc/effective_go.html#channels). (I think it's documented as idiomatic somewhere else too, but this is the first mention I found.)